lms/static/scripts/frontend_apps/components/ContentSelector.tsx

```diff
@@ -179,6 +180,11 @@ export default function ContentSelector({
   const getDefaultValue = (type: DialogType) =>
     type === initialType ? initialValue : undefined;
 
+  const getDefaultValueIfYouTubeUrl = () => {
+    const url = getDefaultValue('url');
+    return url && isYouTubeUrl(url) ? url : undefined;
+  };
```

Comment:
I haven't because the only dependency it has is `getDefaultValue`, which means we need to wrap it in `useCallback` as well to make sure it "updates" every time `initialType` or `initialValue` change.

Do you think it's worth wrapping both in `useCallback`, or leave it like this?

Reply:
> Do you think it's worth wrapping both in useCallback, or leave it like this?

Up to you. The benefit is quite small in absolute terms because the overall UI is not that expensive to render. The React team are currently working on a compiler which adds this stuff automatically, which makes it much less work.

Reply:
> The React team are currently working on a compiler which adds this stuff automatically, which makes it much less work.

Oh, nice. Do you have some link to this?

Reply:
https://react.dev/blog/2022/06/15/react-labs-what-we-have-been-working-on-june-2022#react-compiler. I don't know the latest status, but I did see tweets to the effect that this is a major focus of the team.